### PR TITLE
[HTD-8] Feat: Add `Badge` component

### PR DESCRIPTION
**Pull request by RBoraurito**
# Pull Request Description

## Description

- Add a helper to handle opacity on colors

## Assets

| before | after |
| ------ | ----- |
|        |       |

## Before Merge Checklist

The following steps were checked before merging a PR

- [x] checked and fixed code style issues
- [x] checked and removed unnecessary/junky code
- [x] manually tested introduced solution
- [ ] added needed tests to automate testing this code in the future

### CODE REVIEW

File: src/components/elements/badge/badge.styles.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      case 'secondary':
+        const gradientValue = theme.colors.blueGradient.split(', ');
+        return css`
+          border-radius: 24px;
+          padding: 8px 12px;
+          color: ${({ theme }) => theme.colors.text.alwayLightGrey};
+          background: linear-gradient(180deg, ${gradientValue[0]} 0%, ${gradientValue[1]} 100%);
+          box-shadow:
+            0px 1.6666666269302368px 0.8333333134651184px 0px rgba(201, 212, 251, 0.1) inset,
+            0px -0.8333333134651184px 0.8333333134651184px 0px rgba(16, 46, 151, 0.1) inset;
```
Comment (nephlin7) [inline src/components/elements/badge/badge.styles.ts]: Do we need these long values, can we use max 2 or 4 decimals?

File: src/components/elements/badge/index.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  iconWeight = 'fill',
+}: Component<BadgeProps>) => {
+  const hasAdditionalText = variant === 'secondary' && additionalText;
+
+  return (
+    <BadgeWrapper>
+      {hasAdditionalText && <AdditionalText>{additionalText}</AdditionalText>}
+      <StyledBadge variant={variant}>
+        {Icon && <Icon size={16} weight={iconWeight} />}
+        {children}
```
Comment (nephlin7) [inline src/components/elements/badge/index.tsx]: According to the design specs, all tags should be h3. I've tagged u in Figma.

**Review by nephlin7 — CHANGES_REQUESTED** [review]

File: src/components/elements/badge/index.tsx (hunk 2 above)
Comment (RBoraurito) [reply src/components/elements/badge/index.tsx]: Should it just be the text inside the badge?

File: src/components/elements/badge/badge.styles.ts (hunk 1 above)
Comment (RBoraurito) [reply src/components/elements/badge/badge.styles.ts]: Don't think so I'll cut them

**Comment by RBoraurito** [comment]
@nephlin7 This PR was created from  buttons branch, need to merge it after, the same with Icons PR

**Review by nephlin7 — APPROVED** [review]

**Pull request merged**